src/transformers/image_utils.py

```diff
@@ -62,6 +62,159 @@
 class ChannelDimension(ExplicitEnum):
     FIRST = "channels_first"
     LAST = "channels_last"
+    NONE = "none"
+
+
+def _output_wrapper(output):
+    """
+    Utility function to wrap the output of an ImageArray method, returning an equivalent `ImageArray` if it is a NumPy
+    array.
+
+    If the output is a method, we need to wrap it with a decorator to cast its output to an `ImageArray`.
+    """
+
+    def _cast_to_image_object(result):
+        """
+        Casts `result` to an `ImageArray` if it is a NumPy array.
+        """
+        return ImageArray(result) if isinstance(result, np.ndarray) else result
```

Comment:
I've added logic so that it recursively calls _output_wrapper if the result is a list or tuple, and added splitting as an example in the tests https://github.com/huggingface/transformers/pull/24577/commits/a46bcca5a063964f3f34fea3a7720eab7b4a0987